Library/Homebrew/unpack_strategy/tar.rb

```diff
@@ -38,9 +38,11 @@ def extract_to_dir(unpack_dir, basename:, verbose:)
         end
 
         system_command! "tar",
-                        args:    ["xf", tar_path, "-C", unpack_dir],
+                        args:    ["xf", tar_path, "-C", unpack_dir, *extra_args],
```

Comment:
```suggestion
                        args:    ["xof", tar_path, "-C", unpack_dir],
```
So it turns out that the `-o` option (aka `--no-same-owner) is accepted by both BSD and GNU tar.

See https://www.gnu.org/software/tar/manual/html_section/tar_22.html
and https://www.unix.com/man-page/osx/1/tar/

So we don't in fact need `extra_args`, we can simply pass `-o` to `tar` on both Linux and macOS.
`-o` is the default on macOS (unless `-p` is specified).
`-o` is the default on Linux when the user is non-root.

Reply:
So, no sense to have
`extend/unpack_strategy/tar.rb` and
`extend/os/linux/unpack_strategy/tar.rb`
And requirment in last line `require "extend/unpack_strategy/tar"` is undue.

Reply:
Yep, they can go away. This patch will be a single character in the end, adding one `o`. The best kind of patch! Thanks for your work on this issue.

Reply:
Can I delete unused files, created by me, `Library/Homebrew/extend/unpack_strategy/tar.rb` and Library/Homebrew/extend/os/linux/unpack_strategy/tar.rb.
And resolve conversation about these files?

Reply:
Ok see your last comment